lib/SampleService/core/user_lookup.py

```diff
@@ -91,12 +108,20 @@ def invalid_users(self, usernames: Sequence[UserID]) -> List[UserID]:
             return []
         _no_falsy_in_iterable(usernames, 'usernames')
 
-        r = requests.get(self._user_url + ','.join([u.id for u in usernames]),
+        bad_usernames = [u for u in usernames if not self._valid_cache.get(u.id, default=False)]
+        if len(bad_usernames) == 0:
+            return []
+
+        r = requests.get(self._user_url + ','.join([u.id for u in bad_usernames]),
                          headers={'Authorization': self._token})
         self._check_error(r)
         good_users = r.json()
+        for u in bad_usernames:
+            if u.id in good_users:
+                self._valid_cache.set(u.id, True)
+
         # TODO ACL cache
```

Comment:
Drop the comment since this is adding the ACL cache

Reply:
Done